dhis-2/dhis-support/dhis-support-commons/src/main/java/org/hisp/dhis/commons/util/RelationshipUtils.java

```diff
@@ -44,9 +49,7 @@
      */
     public static String generateRelationshipKey( Relationship relationship )
```

Comment:
why is this class needed?

I think we have too many "utils"
* RelationshipUtils
* RelationshipKeySupport

It would be great if these could be in one place/package. The issue with things getting out of sync like we had them before might also one of ease of finding things.

Reply:
We are generating keys in a lot of places outside tracker.
We could remove `RelationshipKeySupport` but then we need to convert tracker domain `Relationship` to api domain `Relationship` every time we need to generate a key

Reply:
Can we merge these two classes? I dont fully get why we need RelationshipUtils and RelationshipKeySupport

Maybe there is a reason these 2 classes have to exist separately. The current names just don't tell me what they are for. I would also not know where to add changes based on the names.

I haven't looked at them in details so I dont have a more concrete suggestion. We can also leave them as is until we have a better idea.

Reply:
We really cant merge them as they have different scopes.
RelationshipKeySupport does the same exact things of RelationshipUtils but on tracker objects.
We could later think of removing `RelationshipUtils` and put the logic inside `org.hisp.dhis.relationship.Relationship` but then I don't think we can do the same for `RelationshipKeySupport` and `org.hisp.dhis.tracker.domain.Relationship` as we are creating the keys using an external relationshipType due to the idScheme mechanism